src/redux/movies.ts

```diff
@@ -43,34 +43,54 @@ const fetchMovies = async ({
   ]
     .filter(Boolean)
     .join('&');
-  const response = await fetch(`http://localhost:4000/movies?${queryParams}`);
+  const response = await fetch(`${URL}/movies?${queryParams}`);
 
   return await response.json();
 };
 
 export const addMovieAction = createAsyncThunk(
   'addMovie',
   async (movie: Movie) => {
-    await fetch('http://localhost:4000/movies', {
+    await fetch(`${URL}/movies`, {
       method: 'POST',
       body: JSON.stringify(movie),
       headers: { 'Content-Type': 'application/json;charset=UTF-8' },
     });
-    const response = await fetch('http://localhost:4000/movies');
+    const response = await fetch(`${URL}/movies`);
 
     return await response.json();
   },
 );
+enum Method {
+  'Put' = 'PUT',
+  'Get' = 'GET',
+  'Post' = 'POST',
+  'Delete' = 'DELETE',
+}
+interface FetchDataParams {
+  url: string;
+  method?: Method;
+  bodyObj?: any;
+  headers?: HeadersInit;
+}
+const fetchJson = async ({ url, method, bodyObj }: FetchDataParams) => {
```

Comment:
I would advise you to move this declaration to separate file, like `api/index.ts`. It will be easier to reuse in future if you want to add some new thunks responsible for API interaction with other entities, also, it will make this file less overloaded.